woocommerce/woocommerce
# [Navigation] Gracefully handle deprecation of classes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a `RemovedDeprecated` class to handle removed Navigation classes. This way, method calls to these classes will not create fatal errors. Instead, use logger and Tracks to alert developers that these classes should be removed.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Using the Snippets plugin add a new Snippet with the following code to simulate a problematic extension trying to call a removed class.
```php
use Automattic\WooCommerce\Admin\Features\Navigation\Menu;

if ( ! function_exists( 'testFatalCausingImpletmentation' ) ) {
	function testFatalCausingImpletmentation() {
		// call static method
		Menu::add_plugin_item();

		// call instance method
		$menu = new Menu();
		$menu->add_plugin_item();
	}
}

testFatalCausingImpletmentation();
```
2. Navigate to WooCommerce > Home. See that no fatals occur.
3. WooCommerce > Status > Logs > plugin-woocommerce. See the logged out message relating to the misuse of removed classes.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Gracefully handle deprecation of Navigation feature classes.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
